src/infrastructure/content/NetworkMessageHandler.java

```diff
@@ -0,0 +1,151 @@
+package infrastructure.content;
+
+import networking.INotificationHandler;
+import org.json.JSONObject;
+import java.util.HashMap;
+import infrastructure.validation.logger.ILogger;
+import infrastructure.validation.logger.LogLevel;
+import infrastructure.validation.logger.LoggerFactory;
+import infrastructure.validation.logger.ModuleID;
+
+/**
+ * NetworkMessageHandler class implements the INotificationHandler interface.
+ * this includes one method i.e @onMessageReceived()
+ * this class is used to notify and to send data to the UI module 
+ * as new user joined or as a user exit or as message sent by a user
+ * 
+ * @author Talha Yaseen
+ */
+public class NetworkMessageHandler implements INotificationHandler {
+	
+	/**
+	 * Creating instance of ContentCommunicator class to access handler map and image map
+	 */
+	private ContentCommunicator cc = ContentFactory.getContentCommunicator();
+	
+	/**
+	* Creating instance of class which implements ILogger interface
+	*/
+	private ILogger logger = LoggerFactory.getLoggerInstance();
+	
+	/**
+	 * Accessing handlerMap hashmap from ContentCommunicator class to use it for sending message to UI
+	 */
+	private HashMap<String, IContentNotificationHandler> handlerMap = cc.getHandlerMap();
+	
+	/**
+	 * Accessing imageMap hashmap from ContentCommunicator class to access image corresponding to username
+	 */
+	private HashMap<String, String> imageMap = cc.getImageMap();
+	
+	/**
+	 * This variable will store the meta field of json string sent by the Networking module
+	 */
+	private String metafield;
+	
+	/**
+	 * This variable will store the username of client
+	 */
+	private String username;
+	
+	/**
+	 * This variable will store the image accessed by imageMap corresponding to username 
+	 */
+	private String userimage;
+	
+	/**
+	* This variable will store a string which will contains log message that to be passed to log method of logger
+	*/
+	private String logMessage;
+	
+	/**
+	 * Getting handler of the UI module to send the data
+	 */
+	private IContentNotificationHandler handler;
+	 
+	/**
+	 * @param String message - JSON string message (Meta fields are newUser or message or userExit)
+	 *
+	 * if meta field is newUser, then remaining field of parameter message would only be username,  
+	 * to send this message to UI removing metafield, adding image and then calling @onNewUserJoined() method
+	 * if meta field is message, then remaining fields of parameter message would be username, message and time, 
+	 * to send this message to UI removing metafield, adding image and then calling @onMessageReceived() method
+	 * if meta field is exitUser, then remaining field of parameter message would only be username, 
+	 * to send this message to UI removing metafield, adding image and then calling @onNewUserJoined() method
+	 * Networking module will be calling this method with json string message to send data to the content module
```

Comment:
Shorten this, no need for the whole logic you are doing. Just write what the field name will be sent for each function called. Little to no explanation is needed.